### DISCUSSION TITLE
Appreciaton thread

### DISCUSSION THREAD

**Discussion by Matth7878** [Bartuzen/qBitController#30] [Show and tell]
Thanks a lot for working on this. It looks really promising ! Right now I am using on Android a mix of transdroid (to easily add torrent) and VueTorrent (to manage because UI is really nice). I feel like your app would allow me to replace them. 😃

Didn't want to open an issue as you're in the process to implement a lot of features but is it possible in file list to have top folder shown so it can be renamed? It's what desktop version does but it looks like none of controller app I use show it. It's useful so you can rename top folder as . to strip it.
Alternatively in desktop when adding a torrent you can also choose content layout but I am not sure it's something exposed via web api? (I don't know any controller doing it)

Anyway thank you ! Can't wait for 1.0 !!! 😉

**Comment by Bartuzen** [comment]
I'm glad for your appreciation, it gives me morer motivation to work on the project 🙂

> have top folder shown so it can be renamed?

Are you talking about option to change name of the folder that contains a torrent's files? If so, this can be done with `Set Location` option that will be available in the next version.

> choose content layout

Do you mean changing a file's priorty, name, etc. before torrent is added? qBittorrent doesn't expose this API so I couldn't add this. Your best option is to disable `Start torrent` option and modify it after torrent is added.

**Reply by Matth7878** [reply]
> > have top folder shown so it can be renamed?
> 
> Are you talking about option to change name of the folder that contains a torrent's files? If so, this can be done with `Set Location` option that will be available in the next version.

I am speaking about torrent having a top folder and all files attached to it. For instance : 

Parent_Top_Folder
    |__ file1
    |__ file2

If set the location to "torrent_target" it will create "Parent_Top_Folder" folder as it is a folder the torrent file specifies. So you would have : torrent_target\Parent_Top_Folder\ and if you changed location you can only change torrent_target folder.
In qbittorrent Parent_Top_Folder is shown when you're looking at torrent content but not torrent_target as it is just it's location. You can rename Parent_Top_Folder like you would for a file and a trick is to rename it as a dot to move up all child directly in your set location. (And Parent_Top_Folder folder would no longer exist)
Actually I think it's maybe because what is listed in most controller / webui app is just files and folders are ignored to avoid having to deal with displaying a tree. So maybe displaying actual content (in tree with folders as items) to reflect how desktop display content is what is needed and so you could rename any folders.

> > choose content layout
> 
> Do you mean changing a file's priorty, name, etc. before torrent is added? qBittorrent doesn't expose this API so I couldn't add this. Your best option is to disable `Start torrent` option and modify it after torrent is added.

No what I mean is when adding a torrent directly on desktop you have a drop down menu allowing to choose torrent content layout. Choices are to respect original, to create a top subfolder or to don't create one. Last option mean that in my previous example it would ignore "Parent_Top_Folder" and create children directly in your set location.

Hope I explained it better. 😅

**Reply by Bartuzen** [reply]
For the first one, I understand what you mean. But isn't that already possible? You can see all folders and files of a torrent in the app. I tried naming a folder `.` in the app, and it removed folder and moved all files inside that folder to parent folder, as you described.

For the second one, I had never used that feature and I didn't even know it was a thing. That's why I didn't understand at first. I think we can set it with `root_folder` property in add torrent API. I will add this until the next version.

**Reply by Matth7878** [reply]
For the first one you see all files when tapping a torrent then going to files tab? Is that something which changed after 0.4? I see only files. I can rename them but don't have folder(s) listed?

**Reply by Bartuzen** [reply]
Ah, I found the problem. Windows uses `\` as file separator while Linux and mac use `/`. I assumed that qBittorrent API will use same separator on all systems, but I was wrong. Currently, you can see files and folders as tree in Linux and mac, but not on Windows. I will fix this issue until next version.
